### ISSUE TITLE
RequestConfig not constructable from other jvm languages

### ISSUE THREAD

**Issue by jackcviers** [xebia-functional/xef#275]
Kotlin data-classes cannot be constructed in Java or Scala. To fix this, the data class constructor must be annotated with `@JvmOverloads`.

Without this it is not possible to create a `PGVectorStore`: https://github.com/xebia-functional/xef/blob/e1561909b1c94fba3d6b9eabc0f5d44e433ccdb4/integrations/postgresql/src/main/kotlin/com/xebia/functional/xef/vectorstores/PostgreSQLVectorStore.kt#L20

https://github.com/xebia-
functional/xef/blob/e1561909b1c94fba3d6b9eabc0f5d44e433ccdb4/core/src/commonMain/kotlin/com/xebia/functional/xef/llm/models/embeddings/RequestConfig.kt#L5-L8

**Comment by sbrunk** [comment]
I just ran into this as well while trying out xef from Scala.

`RequestConfig` wants a (probably synthetically generated?) `kotlin.jvm.internal.DefaultConstructorMarker` as third parameter, which can't be instantiated. Adding `@JvmOverloads` to the constructor did not work for me.

I also tried adding secondary `constructor(model: EmbeddingModel, user: String): this(model, User(user))` but then the Kotlin compiler complains that both constructors have the same JVM signature.

As a workaround, I added a static factory like this. I'm sure there's a better way, but my Kotlin knowledge is still quite limited.

```kotlin
data class RequestConfig(val model: EmbeddingModel, val user: User) {
  companion object {
    @JvmStatic
    fun create(model: EmbeddingModel, userId: String): RequestConfig {
      return RequestConfig(model, User(userId))
    }
    @JvmInline value class User(val id: String)
  }
```

**Comment by jackcviers** [comment]
Make a PR @sbrunk with a test in xef-scala, please!

**Comment by sbrunk** [comment]
@jackcviers done in #285

**Comment by raulraja** [comment]
These changes should be available in [0.0.3-alpha.2](https://github.com/xebia-functional/xef/tree/0.0.3-alpha.2)
